### ISSUE TITLE
[Feature request] multiepisode demos for Heretic

### ISSUE THREAD

**Issue by thom-wye** [fabiangreffrath/crispy-doom#705]
during a short discussion in the heretic demos thread on doomworld on RTA vs demo runs, I was informed that the vanilla demo format could support containing runs of multiple episodes done back to back, which is currently a category on speedrun.com, but it is my understanding it is not really done in the demo community.

if that is the case, I imagine a console parameter something like "-multiep 34521" which would begin the game/recording (if -record is present) in episode 3, and the after M8's tally screen, the wall of text is either skipped or not, and then episode 4 begins with a wand start, and then episodes 5, 2 and 1, with wandstarts at each M1.

the reset button would restart the game/recording to the first episode of the list, in this example, that would be episode 3.

for consideration, the total time, which is already implemented for heretic, would keep running across all the episodes played, giving a clean total run time at the end.

**Comment by fabiangreffrath** [comment]
Very interesting suggestion! Generally, I'd say this is a job for UMAPINFO, but since we don't have that either, I am not opposed to hard-coding that feature into the port. 

However, let's please start with baby steps and implement episode transition in the canonical order first. Parsing an arbitrary long list of episodes from the command line is another beast on its own.

**Comment by thom-wye** [comment]
> Very interesting suggestion! Generally, I'd say this is a job for UMAPINFO, but since we don't have that either, I am not opposed to hard-coding that feature into the port.
> 
> However, let's please start with baby steps and implement episode transition in the canonical order first. Parsing an arbitrary long list of episodes from the command line is another beast on its own.

got it. I have suggested the custom episode order since you usually start the rta run of Heretic on episode 3 for easier resets in case D'Sparil acts up, so it will probably be something runners will be looking for.

**Comment by fabiangreffrath** [comment]
Will we have a tally screen after ExM8 and then continue to E(x+1)M1?

**Comment by thom-wye** [comment]
yeah, I imagine so. I guess we should, to be fair, pose the question of whether we expect someone to use this feature for just a casual playthrough, and hence whether the end of episode text and graphics should display too, or whether this will be just something for the speedrunners, who only really care about the tally screen and then getting into the next episode as fast as possible.

**Comment by thom-wye** [comment]
on second thought, I don't think the wall of text and episode end graphics would be a problem for speedrunning if you could keypress through them

**Comment by fabiangreffrath** [comment]
Of course everything is possible, but we should agree upon it as soon as possible. 

What do you say if we could define loops in the `-multiepisode` command line parameter? As in `-multiepisode 34512:` which would start with episode 3 again once episode 2 is finished.

**Comment by thom-wye** [comment]
I see no reason for why not, but I also can't say I see any purpose in that; a speedrunner will just reset using the reset key and start a fresh demo recording at the end of the run I expect.
